src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,122 @@
 public class StringManipulation implements StringManipulationInterface {
 
+    private String string;
+
     @Override
     public String getString() {
-        return null;
+        return string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (string == null || string.isEmpty()) {
+            return 0;
+        }
+
+        String trimmedString = string.trim();
+        if (trimmedString.isEmpty()) {
+            return 0;
+        }
+
+        // Split the string by whitespace to get individual words
+        String[] words = trimmedString.split("\\s+");
+
+        return words.length;
     }
 
+
+
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
-    }
+    	 if (n <= 0) {
+             throw new IllegalArgumentException("Error: n is less than or equal to 0");
+         } 
+    	 else if (string == null) {
+             throw new NullPointerException("Error: String has not been initialized");
+         } 
+    	 else if (n > string.length()) {
+             throw new IndexOutOfBoundsException("Error: n exceeds string length");
+         }
+    	 
+    	 
+    	StringBuilder result = new StringBuilder();
+        int count = 0;  
+        for (int i = 0; i < string.length(); i++) {
+            char currentChar = string.charAt(i);
+            count++;
+                if (count % n == 0 && maintainSpacing) {
+                result.append(' ');
+            }    
+            if (count % n != 0) {
+                result.append(currentChar);
+            }
+
+
+        }
+        
+        return result.toString();
+        }
+
+
+
+
 
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        if (string == null) {
+        	throw new NullPointerException("Error: string is not yet initialized");
+        }
+        String[] words = string.trim().split("\\s+");
+        if(startWord <= 0 || endWord <= 0) {
+        	throw new IllegalArgumentException("Error: start index and end index is less than 1");
+        }
+
+        else if (startWord > endWord) {
+            throw new IndexOutOfBoundsException("Error: start index is larger than end");
+        }
+
+        else if (endWord > words.length) {
+            throw new IndexOutOfBoundsException("Error: end index is larger than the string length");
```

Comment:
I think you would be able to combine this else if statement with the one from line 79. This will help make the code shorter and will combine the same type of exception. 
Ex: else if (startWord > endWord || endWord > words.length) {
           throw new IndexOutOfBoundsException(); 
      }